src/Stratis.SmartContracts.Executor.Reflection/Loader/ContractAssemblyLoader.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Reflection;
+using CSharpFunctionalExtensions;
+
+namespace Stratis.SmartContracts.Executor.Reflection.Loader
+{
+    /// <summary>
+    /// Loads assemblies from bytecode.
+    /// </summary>
+    /// <para>
+    /// TODO this may return cached assemblies in the future.
+    /// </para>
+    public class ContractAssemblyLoader : ILoader
+    {
+        /// <summary>
+        /// Loads a contract from a raw byte array into an anonymous <see cref="AssemblyLoadContext"/>.
+        /// </summary>
+        public Result<IContractAssembly> Load(ContractByteCode bytes)
+        {
+            // Assembly.Load(byte[]) loads the assembly into a new anonymous AssemblyLoadContext
+            // TODO in the future, we will use a custom AssemblyLoadContext
+            try
+            {
+                Assembly assembly = Assembly.Load(bytes.Value);
+
+                return Result.Ok<IContractAssembly>(new ContractAssembly(assembly));
+            }
+            catch (BadImageFormatException e)
```

Comment:
Curious about this, I'm interested whether this exception handles all possible errors that we could have.
Given that though, gas would be injected before we reach here so it doesn't really affect us.

thinking out loud..

Reply:
I believe this handles all the exceptions that we expect could be thrown from `Assembly.Load`.